tpm/tpm.go

```diff
@@ -64,6 +71,36 @@ func (t *TPM) Open(ctx context.Context) error {
 		return err
 	}
 
+	if t.enableSimulator {
+		sim := simulator.New()
+		if err := sim.Open(ctx); err != nil {
+			return fmt.Errorf("failed opening TPM simulator: %w", err)
+		}
+		at, err := attest.OpenTPM(&attest.OpenConfig{
+			TPMVersion:     attest.TPMVersion20,
+			CommandChannel: sim,
+		})
+		if err != nil {
+			return fmt.Errorf("failed opening attest.TPM: %w", err)
+		}
+		t.attestTPM = at
+		t.rwc = sim
+	} else {
+		if isGoTPMCall(ctx) {
+			rwc, err := open.TPM(t.deviceName)
+			if err != nil {
+				return fmt.Errorf("failed opening TPM: %w", err)
+			}
+			t.rwc = rwc
+		} else {
+			at, err := attest.OpenTPM(t.attestConfig)
+			if err != nil {
+				return fmt.Errorf("failed opening TPM: %w", err)
+			}
+			t.attestTPM = at
+		}
+	}
```

Comment:
Ok, optionally an `open` method in `go:build tpmsimulator` and another in `go:build !tpmsimulator`.

Reply:
Ideally you should be able to enable the simulator from a third-party package too just using the tags. This can also be handled in the open package too.

Reply:
At the moment I do not intend the C TPM simulator in this repo to be used externally. In fact, people trying to use it would have the same issue `go get`-ing the dependency as we had in the first place. Or are you thinking of different use cases?

Reply:
This code works, and I can see the .c and .h files in my `~/go/pkg/mod/...`. I don't know if more complex stuff won't, but if this library provides useful stuff that we can use for testing or while developing on a macOS, I would say we should allow it.

```go
package main

import (
	"fmt"
	"log"

	"github.com/google/go-tpm-tools/simulator"
	"github.com/google/go-tpm/tpm2"
)

func main() {
	sim, err := simulator.Get()
	if err != nil {
		log.Fatal(err)
	}
	defer sim.Close()

	b, err := tpm2.GetRandom(sim, 32)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("%x\n", b)
}
```

Reply:
I moved back to where I started. Just removed the copy for now and rely on the import directly, but the tags are still present. Third party packages should now be able to do `simulator.New()` (from this package). If they vendor their dependencies, it won't work.